test/rules.jl

```diff
@@ -122,6 +124,21 @@ _second(t) = Base.tuple_type_head(Base.tuple_type_tail(t))
         # Test that @scalar_rule and `One()` play nice together, w.r.t broadcasting
         @inferred frule((Zero(), sx, sy), very_nice, 1, 2)
     end
+
+    @testset "complex inputs" begin
+        @test frule((nothing, 1), complex_times, 1) == (1 + 2im, 1 + 2im)
+        @test frule((nothing, im), complex_times, im) == (-2 + im, -2 + im)
+        ctx, back = rrule(complex_times, 1)
+        @test ctx == 1 + 2im
+        ∂self, ∂x = back(1)
+        @test ∂self == NO_FIELDS
+        @test ∂x == 1 - 2im
+        ctx, back = rrule(complex_times, im)
+        @test ctx == -2 + im
+        ∂self, ∂x = back(im)
+        @test ∂self == NO_FIELDS
+        @test ∂x == 2 + im
+    end
```

Comment:
Could this be tested with finite differencing rather than hard-coded checks? This _might_ be one of those rare cases where it makes sense to hard-code the test though...

Reply:
We certainly could test with FD, but we'd need to add FD as a test dependency here. I don't see any downsides to that.